ansible/playbooks/roles/preflight/tasks/main.yml

```
@@ -30,6 +30,27 @@
 
 - include_tasks: check-os.yml
 
+- include_vars:
+    file: roles/common/vars/main.yml
+    name: common_vars
+
+- name: Validate network configuration
+  when:
+    - common_vars.provider == "any"
+    - common_vars.specification.cloud is undefined
+  block:
+  - name: Validate if ansible_default_ipv4.address matches address from inventory
+    when: ansible_default_ipv4 | length > 0
+    assert:
+      that:
+        - ansible_default_ipv4.address == hostvars[inventory_hostname].ansible_host
+      fail_msg:  "ansible_default_ipv4.address is {{ ansible_default_ipv4.address }} but inventory uses ip: {{ hostvars[inventory_hostname].ansible_host }}. Check default routing configuration, read more in troubleshooting document"
+
+  - name: Empty value exception
+    when: ansible_default_ipv4 | length == 0
+    fail:
+      msg: "ansible_default_ipv4 is empty. Check default routing configuration, read more in troubleshooting document"
```

```suggestion
  - name: Assert default route is defined
    assert:
      that: ansible_default_ipv4.address is defined
      fail_msg: ansible_default_ipv4.address is undefined. Check default routing configuration, read more in troubleshooting document
      quiet: true
```

I would move this task before the one above - to assert first the fact is defined, then validate its value.